Nicolas7Chaves/WageWatch
# Build out skeleton

Summary Description: 

This server.js establishes an Express server that provides two endpoints. The GET endpoint at /employees returns a static list of employee objects, each with details such as ID, name, position, yearly salary, and hourly rate. The POST endpoint at /calculate accepts an array of employee IDs and a meeting duration, calculates the total and per-employee cost for the meeting based on their hourly rates, and returns this data. The server is configured to parse JSON request bodies and handle CORS requests, ensuring it can serve a front end located on a different domain.

Testing Instructions:

1. Start the server with npm run dev in your terminal. This should launch the server using nodemon, which will auto-restart on code changes.

2. To test the GET request, use Postman:
Create a new GET request to http://localhost:3001/employees.
Send the request and observe the JSON array of employee data in the response.

3. For the /calculate endpoint, your server expects a POST request with a JSON body containing an array of employee IDs and the meeting duration in hours. Here is how you should set up your request in Postman:

Set the HTTP method to POST.
Enter the URL http://localhost:3001/calculate.
Go to the "Body" tab in Postman.
Select the raw option.
Choose JSON from the dropdown menu that initially displays "Text".
Enter the JSON data in the following structure:
json
Copy code
{
  "employeeIds": [1, 2],
  "meetingDuration": 2.5
}
Make sure to replace [1, 2] with the actual IDs of the employees you want to include in the calculation and 2.5 with the actual duration of the meeting in hours.
